WalletWasabi.Tests/UnitTests/TransactionProcessorTests.cs

```diff
@@ -291,15 +300,21 @@ public void ReceiveWasabiCoinJoinTransaction()
 			Assert.True(coin.IsLikelyCoinJoinOutput);  // because we are spanding and receiving almost the same amount
 		}
 
-		private static TransactionProcessor CreateTransactionProcessor()
+		private async Task<TransactionProcessor> CreateTransactionProcessorAsync([CallerMemberName] string callerName = "")
 		{
 			var keyManager = KeyManager.CreateNew(out _, "password");
 			keyManager.AssertCleanKeysIndexed();
+
+			var txStore = new AllTransactionStore();
+			var dir = Path.Combine(Global.Instance.DataDir, callerName, "TransactionStore");
+			await IoHelpers.DeleteRecursivelyWithMagicDustAsync(dir);
```

Comment:
This is not good. I wrote these tests because UT can be run quickly and are deterministic (never fail for no reason), with this new dependency now we have an integration test. I don't see easy solution.

Reply:
Can you explain what you mean? I don't understand what would make it unreliable. However they're still not integration tests as we define them, they don't use unreliable external dependencies: https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi.Tests/README.md

# Tests

We divide tests into two different categories. **Unit Tests** and **Integration Tests**.  

We define **Unit Tests** as tests that have no **external dependencies**, however **local dependencies** are allowed. **Unit Tests** run on CI.  
We define **Integration Tests** as tests that have **external dependencies**. Because these **external dependencies** are unreliable, **Integration Tests** do not run on CI.  

Examples for **external dependencies** are communication over the Internet, over the Tor network or with hardware wallets.  
Examples for **local dependencies** are the file system, processes that do not interact with external dependencies, like these HWI's version command call or system mutexes.

Reply:
In software development we don't have good definitions for virtually anything, that sucks. However in this point there has been a decent level of agreement that a *unit test* is a test that does not cross the class boundaries, executes the code in an isolated environment so, dependencies used  to be injected and replaced by mock/fake implementations.

Why? because when a unit test fails it is because the the unit under test (and only the unit under test) is broken. So, in case the `TransactionStore` is broken then `TransactionStoreTests` should fail but the `TransactionProcessorTests` should not. This leads us to the undesirable situation where one error in one component makes lots of tests to fail.

Reply:
How about the `KeyManager`, the `ObservableConcurrentHashSet` or NBitcoin's `Money` object? If they fail, then `TransactionProcessorTests` fail, too. Then we should mock them too! Come on that's silly 😄 

https://stackoverflow.com/questions/5357601/whats-the-difference-between-unit-tests-and-integration-tests/5357837#comment36460115_5357837

> unit tests should only depend on the tested implementation unit; they should not depend on external components such as databases, network services, web browser interaction. When such external elements are required, unit tests use mock objects.

You mock when you have something unreliable to work with or you cannot reproduce a scenario.

Reply:
Yes, `KeyManager` should be mocked too. I didn't do it because it doesn't touch the disk.